### PR TITLE
refactor(ext/http): generic abstract listeners

### PR DESCRIPTION
**Pull request by lucacasonato**
Improve abstractions around listeners to support listener + connection network stream combinations not previously possible (for example a listener exposed as a Tcp, creating Unix network streams).

### CODE REVIEW

File: ext/http/request_properties.rs [hunk 1]
```diff
@@ -72,15 +93,13 @@ pub trait HttpPropertyExtractor {
 
 pub struct DefaultHttpPropertyExtractor {}
 
+#[async_trait::async_trait(?Send)]
```
Comment (mmastrac) [inline ext/http/request_properties.rs:96]: 👍

File: ext/http/http_next.rs [hunk 2]
```diff
@@ -915,8 +902,7 @@ where
   let listen_properties_clone: HttpListenProperties = listen_properties.clone();
   let handle = spawn(async move {
     loop {
-      let conn = listener
-        .accept()
+      let conn = HTTP::accept_connection_from_listener(&listener)
```
Comment (mmastrac) [inline ext/http/http_next.rs:905]: I think we could probably turn this functionality into a trait living in ext/net that any listener type can implement. I don't think we need to do it now, however.

File: ext/http/request_properties.rs [hunk 3]
```diff
@@ -1,34 +1,37 @@
+// Copyright 2018-2023 the Deno authors. All rights reserved. MIT license.
```
Comment (mmastrac) [inline ext/http/request_properties.rs:1]: We need to update the copyright checker to ensure this doesn't happen.

**Review by mmastrac — APPROVED** [review]
LGTM. 

I think we can probably go even further with networking/HTTP traits, but this should be sufficient for now.

File: ext/http/http_next.rs (hunk 2 above)
Comment (lucacasonato) [reply ext/http/http_next.rs:905]: Yes, I wanted to refactor this to be as minimal of a change as possible. I also tried with generic connections, but it became too complicated.

Something for a follow up.

**Pull request merged**